src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -421,6 +421,132 @@ void remove_redundant_reorders::run(program& p) {
+                if (prev_dt == output_layout.data_type || fmt_prev != output_layout.format)
+                    continue;
+
+                bool can_fuse_reorder = false;
+                if (next_ptr->is_type<eltwise>() && (prev_simple && next_simple)) {
+                    // If accumulate type of eltwise is impacted from low precision to high precision, do not fuse to avoid accuracy degradation.
+                    if ((data_type_traits::size_of(prev_dt) >= data_type_traits::size_of(next_dt))
+                        || (data_type_traits::is_floating_point(prev_dt) && data_type_traits::is_floating_point(next_dt))) {
+                        can_fuse_reorder = true;
+                    }
+                }
+
+                // - Fuse reorder into Permute: any node -> reorder -> Permute => any node -> Permute
+                // - Fuse reorder into FC     : any node -> reorder -> FC => any node -> FC
+                if ((next_ptr->is_type<permute>() || next_ptr->is_type<fully_connected>()) && (fmt_prev == fmt_next) &&
+                    ((fmt_prev != format::any) && (fmt_next != format::any))) {
+                    can_fuse_reorder = true;
+                }
+
+                if (!can_fuse_reorder)
+                    continue;
+
+                auto output_padded = static_cast<bool>(output_layout.data_padding);
+                auto can_omit_padding = ((output_layout.format == format::b_fs_yx_fsv16 || output_layout.format == format::b_fs_yx_fsv32) &&
+                                        (prev.get_output_layout().format == format::bfyx || prev.get_output_layout().format == format::b_fs_yx_fsv4)) ||
+                                        (output_layout.format == format::b_fs_zyx_fsv16 && prev.get_output_layout().format == format::bfzyx);
+
+                if (output_padded && !can_omit_padding) {
+                    if (prev.get_users().size() != 1)
+                        continue;
+
+                    if (prev.is_type<input_layout>())
+                        continue;
+
+                    prev.merge_output_padding(output_layout.data_padding);
+                }
+
+                node.can_be_optimized(true);
+                LOG_NODE_REMOVAL(node.id());
+                p.extract_and_remove(node);
+
+                if (next_ptr->is_type<fully_connected>())
+                    next_ptr->recalc_output_layout(true);
+            }
+
+            // Fuse reorder into previous node
+            if (prev.is_type<eltwise>() || prev.is_type<fully_connected>()) {
+                bool can_fuse_reorder = false;
+                if (prev.is_type<fully_connected>() && lo.get_optimization_attributes().use_onednn_impls
+                    && output_layout.format == prev.get_preferred_output_fmt()
+                    && output_layout.data_padding == prev.get_output_layout().data_padding) {
+                    auto check_data_types_for_fully_connected = [](data_types in_dt, data_types wei_dt, data_types out_dt) {
```

Comment:
Please leave a comment like: `// TBD: to be refactored to use are_data_types_suitable_for_onednn`